database/TAG1/USER_FUNCTIONS.func.sql

```diff
@@ -0,0 +1,136 @@
+/******************************************************************************
+ *
+ *		addAgent: Add New Function for TAG AGENT
+ *
+ ******************************************************************************/
+
+DROP FUNCTION IF EXISTS addAgent (first_name varchar(30),
+					last_name varchar(30),
+					userName varchar(30),
+					password varchar(30),
+					address varchar(30),
+					city varchar(30),
+					state_cd varchar(6),
+					zip  varchar(10),
+					country varchar(20),
+					email varchar(60),
+					phone  varchar(20),
+					mobile_phone varchar(20),
+					notes varchar(255),
+					office_location_id integer );
+
+
+CREATE OR REPLACE FUNCTION addAgent(	first_name varchar(30),
+					last_name varchar(30),
+					userName varchar(30),
+					password varchar(30),
+					address varchar(30),
+					city varchar(30),
+					state_cd varchar(6),
+					zip  varchar(10),
+					country varchar(20),
+					email varchar(60),
+					phone  varchar(20),
+					mobile_phone varchar(20),
+					notes varchar(255),
+					office_location_id integer ) RETURNS text AS $$
+BEGIN
+	--Can't create a duplicate	
+	IF userName = '' THEN
+		RETURN 'User name must be at least 4 characters';
```

Comment:
Yea. This was meant as error checking, not application logic. But I will switch it to a table constraint on username.  

 

From: nafg [mailto:notifications@github.com] 
Sent: Wednesday, June 17, 2015 4:37 AM
To: smith4312/TAG-Ticket
Cc: smith4312
Subject: Re: [TAG-Ticket] Db1 (#20)

 

In database/TAG1/USER_FUNCTIONS.func.sql <https://github.com/smith4312/TAG-Ticket/pull/20#discussion_r32605108> :

> +                                   userName varchar(30),
> +                                   password varchar(30),
> +                                   address varchar(30),
> +                                   city varchar(30),
> +                                   state_cd varchar(6),
> +                                   zip  varchar(10),
> +                                   country varchar(20),
> +                                   email varchar(60),
> +                                   phone  varchar(20),
> +                                   mobile_phone varchar(20),
> +                                   notes varchar(255),
> +                                   office_location_id integer ) RETURNS text AS $$
> +BEGIN
> +     --Can't create a duplicate     
> +     IF userName = '' THEN
> +            RETURN 'User name must be at least 4 characters';

I'm personally not a big fan of putting logic in the database 

—
Reply to this email directly or view it on GitHub <https://github.com/smith4312/TAG-Ticket/pull/20/files#r32605108> .  <https://github.com/notifications/beacon/ALe6bCWHHVKoJmAlijnCeJZ0Bd_GpguKks5oUSkagaJpZM4FEcWB.gif>